dev/io.openliberty.threading.virtual.internal/src/io/openliberty/threading/virtual/internal/VirtualThreadOperations.java

```diff
@@ -34,6 +38,21 @@
 @SatisfyingConditionTarget("(&(" + Condition.CONDITION_ID + "=" + JavaInfo.CONDITION_ID + ")(" + JavaInfo.CONDITION_ID + ">=21))")
 public class VirtualThreadOperations implements VirtualThreadOps {
 
+    private ThreadTypeOverride overrideService;
+
+    @Reference(
+               cardinality = ReferenceCardinality.OPTIONAL,
+               policy = ReferencePolicy.DYNAMIC,
+               policyOption = ReferencePolicyOption.GREEDY,
+               unbind = "unsetOverrideService")
+    protected void setOverrideService(ThreadTypeOverride vtos) {
+        overrideService = vtos;
+    }
+
+    protected void unsetOverrideService(ThreadTypeOverride vtos) {
+        overrideService = null;
+    }
```

Comment:
This seems unlikely, but if a ThreadTypeOverride service was already assigned and OSGi found a better match with higher service.ranking, is there a risk that it would do:
```
virtualThreadOperations.setOverrideService(betterMatchingService);
virtualThreadOperations.unsetOverrideService(original);
```
in which case your unset method would leave a `null` in place of the `betterMatchingService`.
If that is correct about how it works, then you would need code like this:
```suggestion
    protected void unsetOverrideService(ThreadTypeOverride vtos) {
        if (overrideService == vtos)
            overrideService = null;
    }
```

Reply:
latest commit resolves this